examples/apps/ai-collab/src/components/TaskGroup.tsx

```diff
@@ -48,6 +52,63 @@ export function TaskGroup(props: {
 	const [isDescriptionEditing, setIsDescriptionEditing] = useState(false);
 	const [isDiffModalOpen, setIsDiffModalOpen] = useState(false);
 
+	/**
+	 * Undo / Redo stacks of {@link Revertible} objects maintained for local branches.
+	 */
+	const [undoStack, setUndoStack] = useState<Revertible[]>([]);
+	const [redoStack, setRedoStack] = useState<Revertible[]>([]);
+
+	useEffect(() => {
+		function onDispose(disposed: Revertible): void {
+			setRedoStack((currentRedoStack) => {
```

Comment:
I'm a bit confused by what's happening here. There is a chance that the function returns `newUndoStack` and that would overwrite the current _redo_ stack. Is that deliberate? In general, I'm not super clear on when would a revertible be disposed and thus what kind of handling we should be doing here.

Reply:
I don't think overwrite is going to happen? I've changed the logic so the intent might be clearer now. 

On **when** the revertible should be disposed, it is up to the user to dispose the revertible object. However, for simplicity (at least in the example app and as you suggested) the app automatically disposes the revertible objects once there are more than 20 instances in the stack.